docs/api-docs/headless/end-to-end-guide/graphql.mdx

```diff
@@ -0,0 +1,756 @@
+
+<Callout type="info">
+While the server-side API clients can use either a Storefront API token or a customer impersonation token, client-side web browsers are not allowed with customer impersonation tokens.
+</Callout>
+
+### Create a customer impersonation token
+
+A customer impersonation token permits you to perform actions as any shopper. To generate a [customer impersonation token](/docs/rest-authentication/tokens#customer-impersonation-tokens), send a POST request to `/v3/storefront/api-token-customer-impersonation`.
+
+<Tabs items={['Request', 'Response']}>
+  <Tab>
+ ```http filename="Example request: Create a customer impersonation token"
+POST https://api.bigcommerce.com/stores/{STORE_HASH}/v3/storefront/api-token-customer-impersonation
+X-Auth-Token: {ACCESS_TOKEN}
+Accept: application/json
+Content-Type: application/json
+
+{
+  "channel_id": 1, 
+  "expires_at": 1602288000 
+}
+```
+</Tab>
+  <Tab>
+
+```json filename="Example response: Create a customer impersonation token"
+  {
+    "data":
+      {
+        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+      }
+      "meta": {}
+  }
+```
+</Tab>
+</Tabs>
+
+In the GraphQL Storefront Playground, enter the customer impersonation token in the **Headers** tab as follows:
+
+```
+{
+  "Authorization": "Bearer {customer impersonation token}"
+}
+```
+
+## Querying product data
+
+The GraphQL Storefront API lets you retrieve product data from a store. The following example demonstrates how to query a simple product using the GraphQL Storefront API. For more information and examples, see the [Guide to Working with Products](/docs/storefront/graphql/examples/products).
+
+<Callout type="info">
+#### Query best practices
+* Query less than 50 resources at a time.
+* Limit query complexity where possible.
+* Limit pagination where possible.
+* Employ security practices and limit scope where possible.
+* Employ caching where possible.
+</Callout>
+
+```graphql filename="Get a product" showLineNumbers copy
+# This query retrieves one product.
+query SingleProduct {
+  site {
+    product (entityId: ${params.id}) {
+      id
+      entityId
+      name
+      sku
+      description
+      prices {
+        price {
+          value
+          currencyCode
+        }
+      }
+    }
+  }
+}
+```
+
+## Creating a cart
+
+BigCommerce's GraphQL Storefront API provides the same cart and checkout objects as the REST Storefront API. The following example shows how to create a new cart by adding a simple product.
+
+<Tabs items={['Request', 'Response']}>
+  <Tab>
+    ```graphql filename="Example mutation: Create a cart (simple)" showLineNumbers copy 
+    # Creates a new cart, adding a simple product.
+    # In the GraphQL Playground, this will fail if you already have a cart.
```

Comment:
"In the GraphQL Playground, this will fail if you already have a cart."

I don't think this is true, though, because you established at the beginning that we're using a customer impersonation token, not a standard token.  Any storefront session you have in progress in your browser will be irrelevant to a request with a customer impersonation token.